packages/langchain-graph-retriever/src/langchain_graph_retriever/adapters/astra.py

```diff
@@ -107,13 +108,25 @@ def with_user_filters(
         ) -> dict[str, Any]:
             return filter if encoded else codec.encode_filter(filter)
 
+    def process_value(v: Any) -> Any:
+        if isinstance(v, immutabledict):
+            return dict(v)
+        else:
+            return v
+
     for k, v in metadata.items():
         for v_batch in batched(v, 100):
-            batch = list(v_batch)
-            if len(batch) == 1:
-                yield (with_user_filters({k: batch[0]}, encoded=False))
+            batch = [process_value(v) for v in v_batch]
+            if isinstance(batch[0], dict):
+                if len(batch) == 1:
+                    yield with_user_filters({k: {"$all": [batch[0]]}}, encoded=False)
```

Comment:
When we encounter this case, we need to do a separate astra query for each case. This will likely affect performance.